petab/visualize/plotter.py

```diff
@@ -669,7 +670,7 @@ def check_data_to_plot(
             """
             contains_inf = False
             max_finite_cond, min_cond = None, np.inf
-            if data_to_plot is not None:
+            if data_to_plot is not None and len(data_to_plot.conditions):
```

Comment:
Data is always associated with a condition, right? Why might there be data and no conditions?

Reply:
> Data is always associated with a condition, right? 

Right. 

> Why might there be data and no conditions?

In the current implementation, `data_to_plot` can be empty, i.e. not contain any datapoints / conditions (without being `None`). 
Overall, this change makes it easier to share a visualization file between different problems. If some measurements are missing, this will just create an empty plot instead of raising some non-informative exception.